plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts

```diff
@@ -0,0 +1,274 @@
+    this.logger.info(
+      `Started indexing ${documents.length} documents for index ${type}`,
+    );
+    const startTimestamp = process.hrtime();
+    const alias = this.constructSearchAlias(type);
+    const index = this.constructIndexName(type, `${Date.now()}`);
+    try {
+      const aliases = await this.elasticSearchClient.cat.aliases({
+        format: 'json',
+        name: alias,
+      });
+      const removableIndices = aliases.body.map(
+        (r: Record<string, any>) => r.index,
+      );
+
+      await this.elasticSearchClient.indices.create({
+        index,
+      });
+      const result = await this.elasticSearchClient.helpers.bulk({
+        datasource: documents,
+        onDocument() {
+          return {
+            index: { _index: index },
+          };
+        },
+        refreshOnCompletion: index,
+      });
+
+      this.logger.info(
+        `Indexing completed for index ${type} in ${duration(startTimestamp)}`,
+        result,
+      );
+      await this.elasticSearchClient.indices.updateAliases({
+        body: {
+          actions: [
+            { remove: { index: this.constructIndexName(type, '*'), alias } },
+            { add: { index, alias } },
+          ],
+        },
+      });
+
+      this.logger.info('Removing stale search indices', removableIndices);
+      if (removableIndices.length) {
+        await this.elasticSearchClient.indices.delete({
+          index: removableIndices,
+        });
+      }
+    } catch (e) {
+      this.logger.error(`Failed to index documents for type ${type}`, e);
+      const response = await this.elasticSearchClient.indices.exists({
+        index,
+      });
+      const indexCreated = response.body;
+      if (indexCreated) {
+        this.logger.info(`Removing created index ${index}`);
+        await this.elasticSearchClient.indices.delete({
+          index,
+        });
+      }
+    }
+  }
+
+  async query(query: SearchQuery): Promise<SearchResultSet> {
+    const { elasticSearchQuery, documentTypes } = this.translator(query);
+    const queryIndices = documentTypes
+      ? documentTypes.map(it => this.constructSearchAlias(it))
+      : this.constructSearchAlias('*');
+    try {
+      const result = await this.elasticSearchClient.search({
+        index: queryIndices,
+        body: elasticSearchQuery,
+      });
+      return {
+        results: result.body.hits.hits.map((d: ElasticSearchResult) => ({
+          type: d._index.split('__')[0],
+          document: d._source,
+        })),
+      };
+    } catch (e) {
+      this.logger.error(
+        `Failed to query documents for indices ${queryIndices}`,
+        e,
+      );
+      return Promise.reject({ results: [] });
+    }
+  }
+
+  private constructIndexName(type: string, postFix: string) {
+    return `${this.indexPrefix}${type}-index__${postFix}`;
+  }
+
+  private constructSearchAlias(type: string) {
+    return `${this.indexPrefix}${type}__${this.aliasPostfix}`;
```

Comment:
```suggestion
    return `${this.indexPrefix}${type}${this.aliasPostfix ? `__${this.aliasPostfix}` : ''}`;
```

I can get this working with our existing ES instance if we modify to only include the `__` delimiter when an `aliasPostfix` is provided at all.  Would you be amenable to a change like that?

Reply:
Hmm, that's interesting. Is that some kind of limitation on the naming in older ES instances?

I'll modify this to be so regardless, indices have hardcoded `index` keyword already so we shouldn't be running into conflicts.